tests/std/tests/P2446R2_views_as_rvalue/test.cpp

```diff
@@ -401,6 +401,17 @@ void test_example_from_p2446r2() {
     assert(ranges::all_of(words, ranges::empty)); // all strings from words are empty (implementation assumption)
 }
 
+// LWG-4083 "views::as_rvalue should reject non-input ranges"
+struct OutputRvalueIterator {
+    using difference_type = int;
+    int operator*() const;
+    OutputRvalueIterator& operator++();
+    void operator++(int);
+};
+using OutputRvalueRange = decltype(ranges::subrange{OutputRvalueIterator{}, unreachable_sentinel});
+
+static_assert(!CanViewAsRvalue<OutputRvalueRange>);

```

Comment:
There are EDG test failures:

```
"D:\build\out\inc\ranges", line 1527: error: template constraint not satisfied
          concept _Can_as_rvalue = requires(_Rng&& __r) { as_rvalue_view{static_cast<_Rng&&>(__r)}; };
                                                          ^
"D:\build\out\inc\utility", line 187: note: atomic constraint not constant
  concept _Different_from = !same_as<remove_cvref_t<_Ty1>, remove_cvref_t<_Ty2>>;
                            ^
          detected during:
            instantiation of
                      "std::_Choice_t<std::ranges::views::_As_rvalue_fn::_St>
                      std::ranges::views::_As_rvalue_fn::_Choose<_Rng>()
                      noexcept [with
                      _Rng=std::ranges::subrange<OutputRvalueIterator,
                      std::unreachable_sentinel_t,
                      std::ranges::subrange_kind::unsized>]" at line 1545
            instantiation of "const
                      std::_Choice_t<std::ranges::views::_As_rvalue_fn::_St>
                      std::ranges::views::_As_rvalue_fn::_Choice [with
                      _Rng=std::ranges::subrange<OutputRvalueIterator,
                      std::unreachable_sentinel_t,
                      std::ranges::subrange_kind::unsized>]" at line 413 of
                      "D:\a\_work\1\s\tests\std\tests\P2446R2_views_as_rvalue\t
                      est.cpp"
```

As a reminder, when changing a test, you should always run that test at a minimum.

Reply:
Oops... I found that EDG accepts this if we switch to use direct-initialization.
```diff
diff --git a/stl/inc/ranges b/stl/inc/ranges
index 6f7e4d06..094aa1d5 100644
--- a/stl/inc/ranges
+++ b/stl/inc/ranges
@@ -1524,7 +1524,7 @@ namespace ranges {
 
     namespace views {
         template <class _Rng>
-        concept _Can_as_rvalue = requires(_Rng&& __r) { as_rvalue_view{static_cast<_Rng&&>(__r)}; };
+        concept _Can_as_rvalue = requires(_Rng&& __r) { as_rvalue_view(static_cast<_Rng&&>(__r)); };
 
         class _As_rvalue_fn : public _Pipe::_Base<_As_rvalue_fn> {
         private:
```

Reply:
Reduced and reported DevCom-10698021.